### DISCUSSION TITLE
Make it easier to add newline in the middle of a hyperlink (add option to disable open page on Enter OR always break line with Ctrl+Enter of Shift+Enter)

### DISCUSSION THREAD

**Discussion by hsandt** [zim-desktop-wiki/zim-desktop-wiki#2514] [Ideas]
I have a feature request but I see two ways to solve the issue so I'd like to discuss it first.

**The issue**

Sometimes I accidentally combine two URLs by removing a newline or pasting a URL right after another, creating a giant incorrect hyperlink: `https://www.example.com/https://www.example.com/`

So I move the caret between the two original URLs (here, right after the first `.com/`) and press Enter, but Zim will open the link when you press Enter in the middle of a URL.

So everytime I need to Undo my mistake and repaste the URL properly, or select the 2nd URL, cut and paste it elsewhere... It would be simpler if I could just insert a new line / break the line between the two URLs.

Variant of the issue; sometimes I accidentally remove a newline before unrelated text (not even a URL) or paste text at the end of a URL, in which case the long string still starts with `http...` so I have the same issue, e.g. `https://www.example.com/unrelated-text`

Note: this applies to all hyperlinks including page links, not just Internet URLs.

**Solutions**

A. Add an option to disable Go to link when pressing Enter entirely

This would fix the issue for users who don't rely on this input, as Enter would always add a newline; but for users who like this input, it may be cumbersome to disable it entirely.

B. Add alternative mappings such as Ctrl+Enter and Shift+Enter to indicate precise behavior: follow link or add new line

For instance, we could keep Enter to follow link if inside a hyperlink, and map Shift+Enter to always add a newline.
Or we could make Ctrl+Enter follow link if inside a hyperlink, and let Enter to always add a newline.

Give your thoughts so we can choose a solution and open a feature request.

**Comment by vossenjp** [comment]
This bites me from time to time as well.  I don't find CTRL-Z, enter, CTRL-V that big a deal as a work-around.  That said, CTRL-Enter == always add newline no matter what seems reasonable too.

Not a big deal for me though, I really only commented to say you're not alone.  :-)
